sys/progress_bar/progress_bar.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       Progress bar implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include "progress_bar.h"
+
+void progress_bar_print(char *prefix, char *suffix, uint8_t value)
+{
+    if (value > 100) {
+        value = 100;
+    }
+
+    /* Hide cursor */
+    printf("\033[?25l");
+
+    /* Move cursor at the beginning of the line */
+    putchar('\r');
+
+    /* Display progress bar prefix if any */
+    if (prefix[0]) {
```

Comment:
The pointers need to be checked (`if (prefix) ...`).

I think it is valid to require nul terminated strings, if the pointer is non-null.
Most users will to ```progress_bar_print("download", NULL, percent);```.
In that case, just the `printf("%s", prefix)` is valid.